chain/tendermint/src/codec.rs

```diff
@@ -0,0 +1,149 @@
+#[path = "protobuf/fig.tendermint.codec.v1.rs"]
+mod pbcodec;
+
+pub use pbcodec::*;
+
+use graph::blockchain::Block as BlockchainBlock;
+use graph::{blockchain::BlockPtr, prelude::BlockNumber};
+
+use std::convert::TryFrom;
+
+impl EventList {
+    pub fn block(&self) -> &EventBlock {
+        self.new_block.as_ref().unwrap()
+    }
+
+    pub fn block_id(&self) -> &BlockId {
+        self.block().block_id()
+    }
+
+    pub fn header(&self) -> &Header {
+        self.block().header()
+    }
+
+    pub fn events(&self) -> Vec<Event> {
+        self.begin_block_events()
+            .into_iter()
+            .chain(self.tx_events().into_iter())
+            .chain(self.end_block_events().into_iter())
+            .collect()
```

Comment:
Here all of the new `Vec` allocations could be turned into `Iterator` ones.

So you don't have to create three vectors and destroy them to create a fourth one. You can just iterate over three and allocate/create one vector.